library/src/extra/rocsparse_spgeam.cpp

```diff
@@ -115,47 +351,44 @@ namespace rocsparse
             {
                 if(mat_C == nullptr)
                 {
-                    RETURN_IF_ROCSPARSE_ERROR(
-                        rocsparse::csrgeam_allocate_descr_memory_template(handle,
-                                                                          mat_A->rows,
-                                                                          mat_B->cols,
-                                                                          alpha,
-                                                                          mat_A->nnz,
-                                                                          beta,
-                                                                          mat_B->nnz,
-                                                                          descr));
+                    RETURN_IF_ROCSPARSE_ERROR(rocsparse::csrgeam_allocate_descr_memory(handle,
+                                                                                       mat_A->rows,
+                                                                                       mat_B->cols,
+                                                                                       alpha,
+                                                                                       mat_A->nnz,
+                                                                                       beta,
+                                                                                       mat_B->nnz,
+                                                                                       descr));
                 }
 
-                RETURN_IF_ROCSPARSE_ERROR(rocsparse::csrgeam_record_descr_alpha_beta_template(
-                    handle, mat_A->rows, mat_B->cols, alpha, mat_A->nnz, beta, mat_B->nnz, descr));
+                descr->record_if_multiplying_by_alpha_beta(alpha, mat_A->nnz, beta, mat_B->nnz);
```

Comment:
That may not be the behaviour we want. I think that should be the user's responsibility. Moreover, do we check for alpha and beta consistency in the compute phase if that's the case?